### DISCUSSION TITLE
Shell commands vs. "fg"

### DISCUSSION THREAD

**Discussion by smurfix** [kovidgoyal/kitty#6676] [Q&A]
Hello,

when I interrupt my editor and then resume it with "fg", surprise! Kitty retitles the window to "fg".

That's … less than helpful. Is it possible to teach it about the actual command? Methinks that this should be discoverable, possibly with some imperfect heuristics, but I'll take the 90% solution over a 0% situation.

**Comment by kovidgoyal** [comment]
Get your editor to send the escape code to set the window title when it
is resumed via SIGCONT.

**Comment by smurfix** [comment]
The problem is that, while I can teach this to my editor, there's "less" and "aptitude" and various other tools where that's not really feasible.

**Comment by kovidgoyal** [comment]
Well, then the solution is to teach it to fg. In your shell config make
fg a function that first sets the title and then actually calls the
shells builtin fg.
